src/undo.h

```diff
@@ -73,13 +73,15 @@ class CBlockUndo
 {
 public:
     std::vector<CTxUndo> vtxundo; // for all but the coinbase
+    std::map<uint256, std::map<int, CAmount> > msc_iaundo;
```

Comment:
As I understood, `msc_iaundo` is always a snapshot of `ScInfo.mImmatureAmounts`. So it's easy to do rollbacks.
What about storing just changes between 2 states (views), like what to add and what to remove? This will lead to changes in `ApplyMatureBalances` and `RestoreImmatureBalances`, but should occupy (in general) less memory than a full snapshot.

Reply:
That would occupy slightly less memory in the 'undo' db but on the other hand would imply storing 2 states (current and previous) both in memory and in 'sc' db.